source/simulator/parameters.cc

```diff
@@ -265,6 +265,15 @@ namespace aspect
                        "The `Newton Stokes' scheme is deprecated and only allowed for reasons of "
                        "backwards compatibility. It is the same as `iterated Advection and Newton Stokes'.");
 
+    prm.declare_entry ("Nonlinear solver failure strategy", "abort program",
+                       Patterns::Selection("continue with next timestep|cut timestep size|abort program"),
+                       "Select the strategy on what to do if the nonlinear solver scheme fails to "
+                       "converge. The options are:\n"
+                       "`continue with next timestep`: ignore error and continue to the next timestep\n"
+                       "`cut timestep size`: reduce the current timestep size by a factor of 2 and redo "
```

Comment:
I think it would be better to not hard-code the factor of 2, because it is an input parameter of that plugin. Please update the documentation.